### PR TITLE
Other small models

### PR DESCRIPTION
**Pull request by digeff**
This PR has this previous PR as a requirement/dependency: https://github.com/Microsoft/vscode-chrome-debug-core/pull/404

SupportedDomains: Class to query domain capabilities
PauseOnExceptionOrRejection: Configure the behaviour when exceptions are thrown or promises rejected
PauseOnExceptionRequestHandlers: Exceptions and promises related client requests
SkipFilesLogic: Implements the Skip files feature
SmartStepLogic: Implements the smart step feature
DotScriptCommand: Implements the .script debugging command
NotifyClientOfLoadedSources: Gives the client updates about which sources are associated with the loaded scripts
SourceRequestHandler: Source related client requests
AsyncStepping: Handles async stepping in the debuggee
SyncStepping: Sync stepping features

### CODE REVIEW

File: src/chrome/internal/domains/supportedDomains.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+export interface ISupportedDomains {
+    isSupported(domainName: string): boolean;
+}
+
+@injectable()
+export class SupportedDomains implements IComponentWithAsyncInitialization, ISupportedDomains {
+    private readonly _domains = new Map<string, CDTP.Schema.Domain>();
+
+    constructor(private readonly _dependencies: ISupportedDomainsDependencies) { }
```
[the remark this answers is not in the document]
Comment (digeff) [reply src/chrome/internal/domains/supportedDomains.ts]: Yes, fixing...

File: src/chrome/cdtpDebuggee/features/cdtpSchemaProvider.ts [hunk 2]
```diff
@@ -2,7 +2,10 @@
  * Copyright (C) Microsoft Corporation. All rights reserved.
  *--------------------------------------------------------*/
 
- import { Protocol as CDTP } from 'devtools-protocol';
+import { Protocol as CDTP } from 'devtools-protocol';
+export interface ISchemaProvider {
+    getDomains(): Promise<CDTP.Schema.Domain[]>;
+}
 
 export class CDTPSchemaProvider {
```
Comment (roblourens) [inline src/chrome/cdtpDebuggee/features/cdtpSchemaProvider.ts]: implements ISchemaProvider?

File: src/chrome/internal/features/smartStep.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+export interface ISmartStepLogicConfiguration {
+    isEnabled: boolean;
+}
+
+export interface IShouldStepInToAvoidSkippedSourceDependencies {
+    stepIntoDebugee(): Promise<void>;
+}
+export class ShouldStepInToAvoidSkippedSource extends BaseActionToTakeWhenPaused {
+    private readonly _dependencies: IShouldStepInToAvoidSkippedSourceDependencies;
```
Comment (roblourens) [inline src/chrome/internal/features/smartStep.ts]: Should this also come from DI?

File: src/chrome/cdtpDebuggee/features/cdtpSchemaProvider.ts (hunk 2 above)
Comment (digeff) [reply src/chrome/cdtpDebuggee/features/cdtpSchemaProvider.ts]: Fixed

File: src/chrome/internal/features/smartStep.ts (hunk 3 above)
Comment (digeff) [reply src/chrome/internal/features/smartStep.ts]: Yes, fixed :)

**Pull request merged**